superset-frontend/src/SqlLab/components/TemplateParamsEditor.jsx

```diff
@@ -42,23 +42,15 @@ const StyledConfigEditor = styled(ConfigEditor)`
   }
 `;
 
-export default class TemplateParamsEditor extends React.Component {
-  constructor(props) {
-    super(props);
-    const codeText = props.code || '{}';
-    this.state = {
-      codeText,
-      parsedJSON: null,
-      isValid: true,
-    };
-    this.onChange = this.onChange.bind(this);
-  }
-
-  componentDidMount() {
-    this.onChange(this.state.codeText);
-  }
+export const TemplateParamsEditor = props => {
```

Comment:
also, it might be better to make this:
`function TemplateParamsEditor(props) {`

so that it's not an anonymous function in stack traces

Reply:
@etr2460 just an FYI, this doesn't apply to react components. React is able to add the component name to stack traces based on the variable name.

Reply:
woah, when did this happen? I could've sworn this wasn't the case....

cc @williaster who I think told me this before too

Reply:
Seems like I misunderstood in the past, it's only if you directly `export default` a function where this becomes an issue. Whichever folks prefer should be fine here, not sure if we want to have a standard best practice for this type of stuff in superset or not (maybe there's a lint rule we could use to enforce consistency, one way or another)

Reply:
That makes sense. A linting rule to prevent anonymous function default exports would be great! I do slightly prefer this pattern since you can use the `React.FunctionComponent<Props>` type